### ISSUE TITLE
Unable to find gru.pth model

### ISSUE THREAD

**Issue by ahmedtalbi** [xl-sr/CAL#18]
Hello,

thanks for your nice work. 

launching 
**python python_client/driving_benchmark.py -c Town01 -v -n test --corl-2017**
looks for CAL/training/models/gru.pth nevertheless this model is not downloaded within the data zip file. (is there another link for pytorch models?)

**Comment by radeonwu** [comment]
i have the same error message as ahmedtalbi.

**Comment by ahmedtalbi** [comment]
you need to checkout a previous commit, 
commit 36000a2ed08ec2d1f6ae23f9d7ffd164e72d4aba

But it would be nice to have acces to the pytorch model if it works.

**Comment by radeonwu** [comment]
unfortunately, after commit 36000a2, the following command can't run successfully any more,
    python python_client/driving_benchmark.py
as it seems lots of differences in this previous commit. 

@ahmedtalbi could you elaborate in a bit more detail the steps that work for you?

**Comment by ahmedtalbi** [comment]
the Folder in that commit will change to PythonClient as it is using the old version of Carla 0.8.x the new one should be based on the 0.9.X (I am not sure didn't check) I only checked out a previous commit and did what was written in the README file.

**Comment by xl-sr** [comment]
Hi all,

the CARLA version is still 0.8.x.

As @radeonwu already wrote, if you want to run the driving benchmark you need to go back to commit 36000a2 - this commit still uses the Tensorflow model on which the evaluation in the paper is based. The new training code, however, uses PyTorch and a trained PyTorch model will be uploaded soon.

So right now you can either:
- use the Tensorflow model with commit 36000a2
- train a PyTorch model and use the current code

**Comment by ahmedtalbi** [comment]
thanks for the answer, 
are you planning to move to version 0.9.x ?

**Comment by xl-sr** [comment]
This is not planned, but I'd be happy about contributions :)

**Comment by AravindChandradoss** [comment]
@xl-sr I got the same error. 

Is PyTorch model available now?  

I tried to checkout 36000a2 but I faced minor import issues (Seems like you guys have changed few directors and file names). Now facing issue in loading model parameters. 
Did you guys @ahmedtalbi @radeonwu  had any issues in "CAL/PythonClient/agents/CAL_agent/perception/model_functions.py"

**Comment by rishikesh1419** [comment]
Hey @xl-sr !
Could you please provide a trained PyTorch model?

**Comment by Marwa215** [comment]
Hi @xl-sr, I have trained the model from scratch with vgg but the predictions weren’t correct it gives the default values e.g -1 for speed limits, red light always=false and hazard stop always =False, i have changed feature extractor and used ResNet18 but still can’t get predictions and loss values for training does not changed after second epoch. I don’t know where is the problem if you have any suggestions could tell me. Also could you proved your trained model. Thanks

**Comment by Marwa215** [comment]
An other question, you mentioned in your paper in the table that different tasks predictions require different input params you trained each task and used toch.cat to concatenate these predictions then trained the whole predictions. I got confused I should train each task block separately or  It is only one training block and how to specify different params for different task blocks during the training? Thanks

**Comment by zzzwww** [comment]
> 
> 
> Hi all,
> 
> the CARLA version is still 0.8.x.
> 
> As @radeonwu already wrote, if you want to run the driving benchmark you need to go back to commit [36000a2](https://github.com/xl-sr/CAL/commit/36000a2ed08ec2d1f6ae23f9d7ffd164e72d4aba) - this commit still uses the Tensorflow model on which the evaluation in the paper is based. The new training code, however, uses PyTorch and a trained PyTorch model will be uploaded soon.
> 
> So right now you can either:
> 
>     * use the Tensorflow model with commit [36000a2](https://github.com/xl-sr/CAL/commit/36000a2ed08ec2d1f6ae23f9d7ffd164e72d4aba)
> 
>     * train a PyTorch model and use the current code

Hi, I am also confused. Does that mean if I want to get the CAL/training/models/gru.pth profile, I need to train the model first? Aren't there any exsiting pth files? Looking forward to your answers. @xl-sr
